### ISSUE TITLE
Option to make the list view more compact

### ISSUE THREAD

**Issue by codemedic** [elementary/files#1219]
## Prerequisites
- [x] I have searched open and closed issues for duplicates.

## Feature
**Problem**
When working with directories containing huge number of files (like a source check out from git etc) the huge space between items is a distraction and makes poor use of the screen real estate.

One could argue that for such use cases, a terminal would be more appropriate; but what about those addicted to nice GUIs from elementary (great work guys)!

**Solution**
Provide an option or an alternate view where the display is more compact.

* Can have tiny icons just enough to identify the file-type
* No thumbnails
* Option to have monospace font would also help (for huge directories with fixed-format filenames)

**Existing work**
Gmail offers a similar approach where it is possible to have the spaced out view (default config), but for those who prefer a compact view, there is an option for that too.

KDE's dolphin can be configured for a compact view.

![image](https://user-images.githubusercontent.com/691011/73788218-8af0f300-4794-11ea-9326-3e857cd40ea2.png)

**Describe alternatives you've considered**
* Switch back KDE, but not as nice over all.
* Use dolphin as a flatpak; dark theme doesn't seem to work.

**Additional context**
N/A

**Comment by aljelly** [comment]
Is something like this what you wanted?

![image](https://user-images.githubusercontent.com/10395308/73849743-54fa5000-4822-11ea-8f38-725208e3d743.png)

Files actually sort of already has this feature (apart from the monospace font). This is the `smallest` view of the list view for Files, however Files' gsettings make this unobtainable by normal means by setting `io.elementary.files.list-view minimum-zoom-level` to `smaller`, so you can't set the list view to any zoom level below that.

It's fortunately not difficult to enable the smallest zoom level for list view. Just run `gsettings set io.elementary.files.list-view minimum-zoom-level smallest` in the Terminal and relaunch Files. Then just scroll down on the mousewheel (or keep pressing <kbd>Ctrl</kbd> <kbd>-</kbd>) until it won't zoom any further. Press <kbd>Ctrl</kbd> <kbd>0</kbd> to reset the zoom to the default.

Possibly the restriction on the `smallest` zoom level could be lifted so you can do this out of the box. I wonder if tiny special 8px emblems could be made specifically for this zoom-level so permissions could still be viewed.

e.g. (mockup, but the 8px emblems are real, edited from the 16px emblem SVGs):

![image](https://user-images.githubusercontent.com/10395308/73853876-790d5f80-4829-11ea-990c-c0dbe178d136.png)

**Comment by KomalleanaByzantis** [comment]
For context, the issue of emblems is discussed in issue #1211. In short, I argue that the smallest zoom level in Column View should show emblems as 12x12 or 8x8 pixel icons. Likewise, List View should have a smaller zoom level, and it should likewise use smaller emblem icons. @aljelly has provided a helpful illustration.

**Comment by jeremypw** [comment]
It would certainly be possible lift the minimum-size restriction on the ListView. It does not seem to have much reason afaict.  It has a reason in IconView as you need a minimum item width for a reasonable label.

Lifting the zoom-level restriction does not solve the empty space problem.  At the moment all spare space is allocated to the Filename column. Another column could be used but there does not seem to be a better choice.  It is not clear how the options listed above under "Solution" would help this aspect either.

A long time ago a PR was submitted for a "compact" view which was like a grid view which showed more information.  However, it did not meet approval. There would have to a strong need to justify supporting the extra code for an additional view type.  Tweaks to the existing views would have more chance of success.

**Comment by jeremypw** [comment]
Note you have the modified column show a constant width date by changing `io.elementary.files.preferences.date-format` to `iso`.

The 'Type' column expands to accomodate the widest type description so there can be a lot if empty space next to the short ones.  Not sure what can be done other than have an option for fixed width columns and ellpsize entries that do not fit.

**Comment by jeremypw** [comment]
Some file managers allow the user to choose the order and content of the columns in ListView which could help making it more compact.

**Comment by codemedic** [comment]
Thanks @aljelly ; while browsing around in the dconf-editor, I figured out that it is possible to have a smaller font. It will be nice to have a more compact, power-user friendly option, configurable from the UI.

Having been a KDE user for several years, I am aware that KDE and its apps have a tendency to over do configurability. It will be nice to have a "canned" option for a compact view with option to set font.

**Comment by JessiAuro** [comment]
The current version of files supports the smallest zoom size now out of box without any gsettings shenanigans.

It seems like the real issue here is that there is no visual indicator that a zoom feature exists. Unless you know about the keyboard shortcuts you're just SOL and that doesn't seem ideal.

I get that the design folks for Files seem to be going for an extreme minimalism, but I'd argue the zoom level is an important enough feature to use UI space for. Some kind of button that opens a popover with a slider perhaps or the `[-][X%][+]` that's all the rage these days.

Having so much functionality hidden behind shortcuts is even more of a problem because it seems like the only official list of them is in a [Github wiki](https://github.com/elementary/files/wiki/shortcuts) amongst a bunch of development docs. Pressing F1 in Files (ironically also keyboard shortcut 😅) just dumps you to the stack exchange.

**Comment by jeremypw** [comment]
@JessiAuro Thanks for the comment - could you raise it as a separate issue (if there is no existing similar issue)?  Then it can be tracked and dealt with separately.  Personally I would not be averse to Files having a limited AppMenu like Code and Terminal have, that includes zoom controls that expose the shortcuts in their tooltips.

**Comment by JessiAuro** [comment]
Yeah I'll look into that. If that's outside the context of this issue then it seems like this issue should be closed too.

**Comment by jeremypw** [comment]
@codemedic I have proposed a PR (https://github.com/elementary/files/pull/2612) that starts to implement a compact monospace view.  So far only the date time column's content is changed but the whole view uses the current default system monospace font if the compact datetime format is selected.
